src/read/pdb/parser.rs

```diff
@@ -119,34 +122,64 @@ where
-                        &chain_id,
-                        (
-                            residue_serial_number + residue_serial_addition,
-                            insertion_code.as_deref(),
-                        ),
-                        (&residue_name, alt_loc.as_deref()),
-                    );
+                    let atom = Atom::new(
+                        hetero,
+                        serial_number + atom_serial_addition,
+                        &name,
+                        x,
+                        y,
+                        z,
+                        occ,
+                        b,
+                        &element,
+                        charge,
+                    )
+                    .expect("Invalid characters in atom creation");
+                    let conformer_id = (residue_name.as_str(), alt_loc.as_deref());
+
+                    let current_chain = if let Some(chain) = current_model.get_mut(&chain_id) {
+                        chain
+                    } else {
+                        current_model.insert(chain_id.clone(), HashMap::new());
+                        current_model.get_mut(&chain_id).unwrap()
+                    };
+
+                    if let Some(residue) = current_chain.get_mut(&(
+                        residue_serial_number + residue_serial_addition,
+                        insertion_code.clone(),
+                    )) {
+                        residue.add_atom(atom, conformer_id);
+                    } else {
+                        current_chain.insert(
+                            (
+                                residue_serial_number + residue_serial_addition,
+                                insertion_code.clone(),
+                            ),
+                            Residue::new(
+                                residue_serial_number + residue_serial_addition,
+                                insertion_code.as_deref(),
+                                Some(
+                                    Conformer::new(
+                                        residue_name.as_str(),
+                                        alt_loc.as_deref(),
+                                        Some(atom),
+                                    )
+                                    .expect("Invalid characters in Conformer creation"),
+                                ),
+                            )
+                            .expect("Invalid characters in Residue creation"),
+                        );
+                    }
+
                     last_residue_serial_number = residue_serial_number;
                     last_atom_serial_number = serial_number;
                 }
                 LexItem::Anisou(s, n, _, _r, _c, _rs, _, factors, _, _e, _ch) => {
                     let mut found = false;
-                    for atom in current_model.atoms_mut().rev() {
+                    for atom in current_model
+                        .values_mut()
```

Comment:
With `IndexMap` this can be changed to `values_mut().rev()` so the benefit from iterating from the end will be retained